### DISCUSSION TITLE
Codesandbox testing example isn't working.

### DISCUSSION THREAD

**Discussion by adriantrunzo** [react-hook-form/react-hook-form#4142] [Q&A] [answered]
Hi all,

I am trying to learn how to write tests for my forms and noticed that the [codesandbox](https://codesandbox.io/s/react-hook-form-unit-test-docs-ewpyt?file=/src/App.test.js) linked from the [testing section of the documentation](https://react-hook-form.com/advanced-usage#TestingForm) isn't working as expected. Specifically, the tests are all failing with messages similar to what's below:

![Screen Shot 2021-02-09 at 9 17 28 AM](https://user-images.githubusercontent.com/1218900/107393112-a3869e00-6ab7-11eb-95b5-a49d510aee3b.png)

I've run into some hard to diagnose issues with my code locally and am hoping that a working sandbox will help me resolve them. Does anybody know how to resolve the issues with the codesandbox above? Perhaps I am not using codesandbox correctly.

Thanks!

**Comment by bluebill1049** [comment, marked as answer]
There is an issue with CSB, clone the code and run them locally.

**Reply by adriantrunzo** [reply]
Okay, thanks for the info. I'll try running the examples locally.

Is the CSB issue documented somewhere that can I learn more about it?

**Reply by bluebill1049** [reply]
It's not resolved.

**Reply by adriantrunzo** [reply]
Oh sorry, maybe I misunderstood what you said. Do you mean that there is an issue with that particular codesandbox example that will be fixed at some point by the react-hook-form team or that there is an issue with the codesandbox app in general? I thought you meant the latter and that the issue was not due to the example code.

**Reply by bluebill1049** [reply]
oh, i meant codesandox had an issue running that test, and we have fired an issue and it's not resolved.

**Reply by adriantrunzo** [reply]
Ok, thanks for the clarification! I appreciate your time.
